akka-persistence/src/test/scala/akka/persistence/PersistentActorJournalProtocolSpec.scala

```diff
@@ -74,7 +74,10 @@ akka.persistence.snapshot-store.plugin = "akka.persistence.no-snapshot-store"
   }
 }
 
-object JournalPuppet extends ExtensionKey[JournalProbe]
+object JournalPuppet extends ExtensionId[JournalProbe] {
+  override def createExtension(system: ExtendedActorSystem): JournalProbe =
+    new JournalProbe()(system)
+}
```

Comment:
see comment in migration guide

Strange that this works. I see that `TestKitExtension` is implemented like this, but in all other places we use `ExtensionIdProvider`. For consistency I think we should use `ExtensionIdProvider`